src/core/linalg/src/sparse/4C_linalg_view.hpp

```diff
@@ -28,6 +29,37 @@ namespace Core::LinAlg
     static_assert("You need to specialize this struct for the wrapper class.");
   };
 
+
+  /**
+   * A wrapper around unique_ptr, which propagates constness to the pointee.
+   * This type is useful to hold wrapped objects in various LinAlg wrappers in combination with
+   * the View mechanism.
+   */
+  template <typename T>
+  struct ConstPropagatingUniquePtr
+  {
+    using Deleter = std::function<void(T*)>;
+
+    ConstPropagatingUniquePtr() = default;
+    template <typename D>
+    ConstPropagatingUniquePtr(std::unique_ptr<T, D>&& obj) : obj(std::move(obj))
+    {
+    }
+
+    ConstPropagatingUniquePtr(T* ptr) : obj(ptr, std::default_delete<T>{}) {}
+    ConstPropagatingUniquePtr(T* ptr, Deleter deleter) : obj(ptr, deleter) {}
+
+    const T& operator*() const { return *obj; }
+    T& operator*() { return *obj; }
+
+    const T* operator->() const { return obj.get(); }
+    T* operator->() { return obj.get(); }
+
+    //! The actual unique_ptr. You may access it directly, but be aware that constness will not
+    //! be propagated this way.
+    std::unique_ptr<T, Deleter> obj;
```

Comment:
Is this really required to be public? I would prevent the direct access to `obj`.

Reply:
Changed.